src/lib.rs

```diff
@@ -697,6 +697,17 @@ where
     fn magnitude2(self) -> Self::Scalar {
         self.clone().dot(self)
     }
+
+    /// Returns the [reflection](https://en.wikipedia.org/wiki/Reflection_(mathematics))
+    /// of the current vector with respect to the given surface normal. The surface normal
+    /// must be of length 1 for the return value to be correct. The current vector is
+    /// interpreted as pointing toward the surface, and does not need to be normalized.
+    fn reflect(self, surface_normal: Self) -> Self {
+        // We can't multiply by a literal 2 or 2.0 value and we don't have a Two interface, but since
```

Comment:
Please add a `Two` interface instead of this hack to prevent weird behaviour when you forget to normalize.

Reply:
Hey I prefer the term "clever workaround" :)

Instead of adding a Two trait, I thought back to my elementary school math class where we learned that 2x = x + x. See latest code for the simpler approach.